packages/lib/server-only/webhooks/zapier/list-documents.ts

```diff
@@ -2,23 +2,38 @@ import type { NextApiRequest, NextApiResponse } from 'next';
 
 import { findDocuments } from '@documenso/lib/server-only/document/find-documents';
 import { getRecipientsForDocument } from '@documenso/lib/server-only/recipient/get-recipients-for-document';
+import type { Recipient, Webhook } from '@documenso/prisma/client';
 
+import { getWebhooksByTeamId } from '../get-webhooks-by-team-id';
 import { getWebhooksByUserId } from '../get-webhooks-by-user-id';
 import { validateApiToken } from './validateApiToken';
 
 export const listDocumentsHandler = async (req: NextApiRequest, res: NextApiResponse) => {
   try {
     const { authorization } = req.headers;
-    const user = await validateApiToken({ authorization });
+    const { user, userId, teamId } = await validateApiToken({ authorization });
 
-    const documents = await findDocuments({ userId: user.id });
-    const allWebhooks = await getWebhooksByUserId(user.id);
-    const recipients = await getRecipientsForDocument({
-      documentId: documents.data[0].id,
-      userId: user.id,
-    });
+    let allWebhooks: Webhook[] = [];
+    let documents;
```

Comment:
This can just be `findDocuments({ userId, teamId, perPage: 1 })` since you only want the first document.

If `teamId` is undefined it won't be used during the filter and saves a bunch of headache.

This also means recipients won't need to be in the if statements either 😄

Reply:
Smart! Thanks!